pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -153,6 +166,10 @@ func NewManager(topology []cadvisorapi.Node, topologyPolicyName string) (Manager
 		return nil, fmt.Errorf("unknown policy: \"%s\"", topologyPolicyName)
 	}
 
+	if topologyScopeName != containerTopologyScope && topologyScopeName != podTopologyScope {
+		return nil, fmt.Errorf("unknown scope: \"%s\"", topologyScopeName)
+	}
+
```

Comment:
To make things cleaner, I'd rather see a `Scope` type which gets instantiated here (similar to the policy right above this) and then invoked as `scope.Admit(pod)` below instead of having all of these string comparisons everywhere (e.g. the code referenced here as well as the code shown below):
```
	if m.scope == podTopologyScope {
		return m.podAdmit(pod)
	}

	return m.containerAdmit(pod)
```

Reply:
I checked such refactoring (moving scopes into `scope.go`, `scope_pod.go`, `scope_container.go`. The scope is instantiated the same as policy and it implements current TM logic. However, the scope must call a policy while gathering/merging hints. 

The scope after refactoring would be in such form:
```
type Scope interface {
	Name() string
	Admit(pod *v1.Pod) lifecycle.PodAdmitResult
	AddHintProvider(h HintProvider)
	GetAffinity(podUID string, containerName string) TopologyHint
	RemoveContainer(containerID string) error
	AddContainer(pod *v1.Pod, containerID string) error
}

type scope struct {
	mutex sync.Mutex
	name string
	//Mapping of a Pods mapping of Containers and their TopologyHints
	//Indexed by PodUID to ContainerName
	podTopologyHints PodTopologyHints
	hintProviders []HintProvider
	policy Policy
	podMap map[string]string
}
```

You can check what I meant below:
https://github.com/cezaryzukowski/kubernetes/pull/1

Please let me know if this is what you meant.